### PR TITLE
Only documentation changes

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Documentation for only changes related to API

### CODE REVIEW

File: docs/rest_api.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+http://BMI_SERVER:PORT/remove_snapshot
+with DELETE request type and request body as:
+
+body:
+
+{
+ "img" : "<img_name>" ,
+ "project" : "<project_name>" ,
+ "snap_name" : "snapshot_name>"
+}
```
Comment (gsilvis) [inline docs/rest_api.md]: DELETE calls are not allowed to have bodies

File: docs/rest_api.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+"snap_name":"test_snap2016" 
+}
+
+This should return a 200 or other errors as explained above.
+
+List snapshots:
+
+We can have as many snapshots as possible for a particular image. So, using this we can list all the snapshots available for an image.
+
+http://BMI_SERVER:PORT/list_snapshots
```
Comment (gsilvis) [inline docs/rest_api.md]: This call does not change any state on the server, so it should be a `GET`

**Comment by gsilvis** [comment]
This API is not using the various REST calls correctly.  Here are the basic rules you should follow:

- For any informational operation that does not change the state of the server, use `GET`
- To create a resource, use `PUT` on a path that will represent the resource: for example, `PUT /images/<new_image_name>`
- Another way to create a new resource is to use a `POST` on the directory that will contain the new resource:  `POST /images`
- To delete a resource, use `DELETE` on the path that represents the resource: `DELETE /images/<image_name>`
- For truly miscellaneous operations (e.g. 'power_cycle_node' in HaaS), use a POST call on a URL somehow linked to any applicable resource; e.g. `POST /images/<image_name>/frobnicate_image`

**Comment by gsilvis** [comment]
But, if the API itself is not up for debate in this PR, then this documentation looks okay

**Comment by rahulait** [comment]
Agree with George. Ideally it should follow the REST semantics than the URLs. Maybe we can fix it now rather than doing in future? Rest looks ok to me.

**Comment by rahulait** [comment]
+1

**Pull request merged**